domain/src/main/java/org/oppia/domain/profile/ProfileManagementController.kt

```diff
@@ -73,10 +225,114 @@ class ProfileManagementController @Inject constructor(
+//          if (updateLoggedInResult.isSuccess()) {
+//            pendingLiveData.postValue(AsyncResult.success(null))
+//          } else if (updateLoggedInResult.isFailure()) {
+//            pendingLiveData.postValue(AsyncResult.failed(updateLoggedInResult.getErrorOrNull()!!))
+//          }
+//        }
+//      } else if (setIdResult.isFailure()) {
+//        pendingLiveData.postValue(AsyncResult.failed(setIdResult.getErrorOrNull()!!))
+//      }
+//    }
+
+    dataProviders.transformAsync("test2", setCurrentProfileId(profileId)) {
+      val deferred = updateLastLoggedIn(profileId)
+      deferred.invokeOnCompletion {
+        if (it != null) {
+          logger.e("ProfileManagementController", "Failed to update last logged in", it)
+          pendingLiveData.postValue(AsyncResult.failed(it))
+        } else {
+          when (deferred.getCompleted()) {
+            ProfileActionStatus.SUCCESS -> pendingLiveData.postValue(AsyncResult.success(null))
+            else -> pendingLiveData.postValue(AsyncResult.failed(ProfileNotFoundException("ProfileId ${profileId.internalId} does not match an existing Profile")))
+          }
+        }
+      }
+      AsyncResult.success(null)
+    }
+    return pendingLiveData
+  }
+
+  /**
+   * Sets the currentProfileId to the selected profile in ProfileChooserFragment.
+   * Checks to ensure that profileId corresponds to an existing profile.
+   *
+   * @param profileId the ID corresponding to the profile being set.
+   * @return a [LiveData] that indicates the success/failure of this set operation.
+   */
+  internal fun setCurrentProfileId(profileId: ProfileId): DataProvider<Any?> {
+    return dataProviders.transformAsync("test", profileDataStore) {
+      if (it.profilesMap.containsKey(profileId.internalId)) {
+        currentProfileId = profileId.internalId
+        return@transformAsync AsyncResult.success<Any?>(null)
+      }
+      AsyncResult.failed<Any?>(ProfileNotFoundException("ProfileId ${profileId.internalId} is not associated with an existing profile"))
+    }
+  }
+
+  /**
+   * Updates the last logged in timestamp of an existing profile.
+   *
+   * @param profileId the ID corresponding to the profile being updated.
+   * @return a [LiveData] that indicates the success/failure of this update operation.
+   */
+  @ExperimentalCoroutinesApi
+  internal fun updateLastLoggedIn(profileId: ProfileId): Deferred<Any?> {
+    return profileDataStore.storeDataWithCustomChannelAsync(updateInMemoryCache = true) {
+      val profile = it.profilesMap[profileId.internalId] ?: return@storeDataWithCustomChannelAsync Pair(it, ProfileActionStatus.PROFILE_NOT_FOUND)
+      val updatedProfile = profile.toBuilder().setLastLoggedInTimestampMs(Date().time).build()
+      val profileDatabaseBuilder = it.toBuilder().putProfiles(profileId.internalId, updatedProfile)
+      Pair(profileDatabaseBuilder.build(), ProfileActionStatus.SUCCESS)
+    }
+//    deferred.invokeOnCompletion {
```

Comment:
Same comment here, can we remove this piece of code, or maybe add some description as to why this needs to be here but commented out

Reply:
Sorry, I think you reviewed my code while I was in the middle of addressing Ben's comments. Commented code was removed.